### PR TITLE
Grip feature

### LINKED ISSUE #114: ARTEMIS - GRIP simple frontend connection
**Is your feature request related to a problem? Please describe.**
Missing connection with CAIDA's grip project https://grip-dev.caida.org/

**Describe the solution you'd like**
1. User clicks on the details of an ARTEMIS hijack
2. Server calls `https://api.grip.caida.org/v1/json/events` using the `event_type`, `asns`, `pfxs` matching the corresponding ARTEMIS hijack fields: `prefix`, `type` (needs mapping from our 4-dimension characterization to the grip keywords), `hijack_as`. I am not sure if these suffice to pinpoint a singular event, will see.
3. It gets back (hopefully :slightly_smiling_face:) details and the ID of the grip event
4. Then it should point dynamically in the rendered page a link like `https://grip-dev.caida.org/events/<type>/<ID>`

**Describe alternatives you've considered**
Link on hijack detail page vs all the info there directly. Will go with the external link for now.

**Additional context**
https://github.com/CAIDA/grip-api/wiki/API-Documentation

### PR DESCRIPTION
**Pull request by CuriouzK0d3r**
### Description of PR

Resolves #114 . In case the GRIP event exists for a specific hijack, we add a redirection button at the hijack page. 

What component(s) does this PR affect?

  - [ ] Back-End (Node.js, Next.js)
  - [x] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
  Does the PR require changes on other components? If yes, please mark the components:
  
  - [ ] Back-End (Node.js, Next.js)
  - [ ] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
### Related Issue
  
  
  
  
  
  
  Resolves #
  
### Type
  
  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [ ] Docs update
  - [ ] None of the above
  
  
  
### Checklist:
  
  - [ ] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
  - [ ] This change requires a change in the documentation.
  - [ ] I have updated the documentation accordingly.
  >)`>>>)>

### CODE REVIEW

File: apps/artemis-web/components/hijack-info/hijack-info.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-info/hijack-info.tsx:65]: what about the hijack type (which should be GRIP-mapped) according to comment https://github.com/FORTH-ICS-INSPIRE/artemis-web/issues/114#issuecomment-884239572 ?

Comment (CuriouzK0d3r) [reply apps/artemis-web/components/hijack-info/hijack-info.tsx]: fixed!

File: apps/artemis-web/pages/hijack.tsx
Comment (vkotronis) [inline apps/artemis-web/pages/hijack.tsx:89]: what does this offer? If we do not have `subscriptionData` then what do `data.view_hijacks` contain actually?

**Review by vkotronis — COMMENTED** [review]
Could you also add in the related issue how this was tested exactly (needed update and config files, etc.)? So that if someone starts with an empty DB, to be able to replicate the PR. Please also add some screenshots from your setup in action where the redirection works (where ARTEMIS and GRIP agree on a hijack and ARTEMIS redirects to the GRIP event)

File: apps/artemis-web/components/hijack-info/hijack-info.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-info/hijack-info.tsx:23]: what is right and left?

File: apps/artemis-web/components/hijack-info/hijack-info.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-info/hijack-info.tsx:417]: this opens up a separate tab, correct? Also, if no GRIP event could be retrieved, what is the visual behavior of this in the hijack info page? (please add screenshot)

Comment (vkotronis) [reply apps/artemis-web/components/hijack-info/hijack-info.tsx:417]: Also, what happens if multiple GRIP events are retrieved? (several IDs I mean). Could you double-check the API call? We can select one of them if needed.

File: apps/artemis-web/components/hijack-info/hijack-info.tsx
Comment (CuriouzK0d3r) [reply apps/artemis-web/components/hijack-info/hijack-info.tsx:23]: The hijack table is split in half to fit in screen.

The left side contains some key-value pairs and the same does the right side

File: apps/artemis-web/components/hijack-info/hijack-info.tsx
Comment (CuriouzK0d3r) [reply apps/artemis-web/components/hijack-info/hijack-info.tsx:417]: interesting thought! I just fetch the first one.

We could add multiple buttons if they are more? Or a dropdown menu?

Comment (vkotronis) [reply apps/artemis-web/components/hijack-info/hijack-info.tsx:417]: drop-down menu would be fine

**Comment by vkotronis** [comment]
> Could you also add in the related issue how this was tested exactly (needed update and config files, etc.)? So that if someone starts with an empty DB, to be able to replicate the PR. Please also add some screenshots from your setup in action where the redirection works (where ARTEMIS and GRIP agree on a hijack and ARTEMIS redirects to the GRIP event)

Hello Alex just a reminder on this to provide:
1. Test files
2. Screenshots
3. Some documentation on how the new feature behaves

File: apps/artemis-web/components/hijack-table/hijack-table.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-table/hijack-table.tsx:147]: this is hijacker, not hijacked

File: apps/artemis-web/components/hijack-table/hijack-table.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-table/hijack-table.tsx:148]: this is hijacker, not hijacked

File: apps/artemis-web/components/hijack-table/hijack-table.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-table/hijack-table.tsx:195]: this is hijacker, not hijacked

File: apps/artemis-web/pages/hijack.tsx
Comment (CuriouzK0d3r) [reply apps/artemis-web/pages/hijack.tsx:89]: the same thing :) its websockets vs rest :)

File: apps/artemis-web/components/hijack-info/hijack-info.tsx
Comment (vkotronis) [inline apps/artemis-web/components/hijack-info/hijack-info.tsx:79]: don't forget to take this out pre-merging

**Comment by vkotronis** [comment]
LGTM! Pending comments are resolved. Only a few things before approval:
1. Test files (whatever you have on announcements and configurations to replicate)
2. Screenshots (with sample events)
3. Some documentation on how the new feature behaves
4. (minor) correct "hijacked AS" to "hijacker AS" in the hijack info page (is OK at the rest occurences)

Thanks Alex! After these, we can merge. You can store the test files in a dedicated folder in the repo, reserved for demos, etc.

**Comment by vkotronis** [comment]
Also what happens when the GRIP API is down? Will the entire hijack info page not render, or just the component?

**Review by vkotronis — APPROVED** [review]
LGTM, I think we can merge. Reminder on my side: add also info to mono-repo ARTEMIS docs

**Comment by vkotronis** [comment]
@CuriouzK0d3r check though failing cypress test first pre-merging

**Comment by CuriouzK0d3r** [comment]
> @CuriouzK0d3r check though failing cypress test first pre-merging

There is a Cypress test that failed randomly :) Its ok now

**Comment by CuriouzK0d3r** [comment]
> frozen until GRIP move is completed

Whats the status with this?

**Comment by vkotronis** [comment]
> > frozen until GRIP move is completed
> 
> Whats the status with this?

No new updates, I think we can proceed to avoid keeping this in the freeze while the master branch changes.
So you can merge if everything is working as expected.

**Pull request merged**
